deepparse/comparer/address_comparer.py

```diff
@@ -0,0 +1,170 @@
+from deepparse.parser import AddressParser
+from typing import List, Union
+
+
+class AdressComparer:
+    """
+        Compares addresses with each others and retrieve the the differences between them.
+    """
+    
+    def __init__(self, parser: AddressParser) -> None:
+        """
+        Address parser used to parse the addresses
+        """
+
+        self.parser = parser
+
+    def __str__(self) -> str:
+        return f"Compare addresses with {self.parser.model_type.capitalize()}AddressParser"
+
+    __repr__ = __str__  # to call __str__ when list of address
+    
+    def compare_with_deepparse(self, addresses_to_compare: Union[List[List[tuple]], List[tuple]]) -> dict:
+        """
+        Compare a list of already parsed addresses and then compare with the result 
+        it would be if it was parsed using deepparse 
+
+        Args:
+            addresses_to_compare (Union[List[List[tuple]], List[tuple]]): List of tuple where
+            the first element in the tuple is the value of the address component, and the 
+            second element is the name of the address component
+
+        Return:
+            Dictionnary that contains dictionnaries that contains all addresses components that differ from the original
+            parsing and the deepparsed components
+        """
+        if isinstance(addresses_to_compare[0], tuple):  # when tag is also the tag and the probability of the tag
+            addresses_to_compare = [addresses_to_compare]
+
+        #rebuilding addresses by joining the values of the address components
+        #and then parse the addresses with deepparse
+        list_of_deepparsed_addresses = []
+        for address in addresses_to_compare:
+            rebuilt_raw_address = " ".join([element[0] for element in address])
+            list_of_deepparsed_addresses.append(self.parser(rebuilt_raw_address))
+
+
+        #get the addresses components from deepparse
+        list_of_lists_of_tuple_of_deepparse_attr = []
+        for deepparsed_address in list_of_deepparsed_addresses:
+            list_of_lists_of_tuple_of_deepparse_attr.append(deepparsed_address.to_list_of_tuples()) 
+
+        #a dictionnary that will contains the delta dict of the parsed addresses to compare
+        dict_of_delta_dicts = {}
+        for index, address_to_compare in enumerate(addresses_to_compare):
+            args_delta_dict = {'deepparse': list_of_lists_of_tuple_of_deepparse_attr[index],
+                                str(index): address_to_compare}
+
+            dict_of_delta_dicts[str(index)] = self.delta_dict_from_dict(args_delta_dict)
+
+        return dict_of_delta_dicts
```

Comment:
I'm not sure of the output. 

I think we need to ask ourselves first if we want to compare one or multiple addresses at the same time. 

If multiple, I think the dict keys should be the reconstructed address or something like that.
An alternative would be to return a list of tuples [(first address as raw input, the diff), (second address as raw input, the diff).

If single address comparison, then we could simply return a tuple or something like that.

Final thoughts, can we return a new object? ComparedAddress()? something like that where we have an attribute .raw_address, diff with us etc. That way, we could simply return a list of those objects and users could manipulate them more easily.

Reply:
Great idea, it would replicate AdressParser and FormattedParsedAddress. it is also easy to understand

Reply:
@MAYAS3, any thoughts? I think the last idea is more interesting in the long run.

Reply:
First of all, I'm really sorry for the delayed response. I forgot about this PR amongst all the Github notifications I received lately.

Secondly, I totally agree with you @davebulaval. The idea is to focus on the diff but to also make the rest of the data (i.e: raw address, user's parsing and deepparse parsing) available. It could be done elegantly with a new object as suggested or by creating operations such as *diif* and *intersect* for FormattedParsedAddress and creating methods in this class to return just what the user wants (with the raw address).